Sources/SwiftGodot/Core/ObjectCollection.swift

```diff
@@ -25,8 +43,9 @@ public class ObjectCollection<Element: Object>: Collection {
         gi.array_set_typed (&array.content, GDExtensionVariantType (GDExtensionVariantType.RawValue(Variant.GType.object.rawValue)), &name.content, &variant.content)
     }
     
+    /// Initializes the collection with an empty typed GArray
     init () {
-        array = GArray ()
+        array = GArray (Element.self)
```

Comment:
Even though this is not public, I added a docstring, and I also changed the line here to initialize a type GArray like VariantCollection does.